docs/source/cpp/streaming_execution.rst

```diff
@@ -175,9 +175,607 @@ their completion::
+are required to create this option::
+
+    // data generator
+    arrow::AsyncGenerator<arrow::util::optional<cp::ExecBatch>> gen() { ... }
+    
+    // data schema 
+    auto schema = arrow::schema({...})
+    
+    // source node options
+    auto source_node_options = arrow::compute::SourceNodeOptions{schema, gen};
+    
+    // create a source node
+    ARROW_ASSIGN_OR_RAISE(arrow::compute::ExecNode * source,
+                            arrow::compute::MakeExecNode("source", plan.get(), {}, 
+                            source_node_options));
+
+``FilterNode``
+--------------
+
+:class:`FilterNode`, as the name suggests, provide a container to define a data filtering criteria. 
+Filter can be written using :class:`arrow::compute::Expression`. For instance if the row values
+of a particular column needs to be filtered by a boundary value, ex: all values of column b
+greater than 3, can be written using :class:`arrow::compute::FilterNodeOptions` as follows::
+
+    // a > 3
+    arrow::compute::Expression filter_opt = arrow::compute::greater(
+                                  arrow::compute::field_ref("a"), 
+																	arrow::compute::literal(3));
+
+Using this option, the filter node can be constructed as follows::																
+
+    // creating filter node
+    arrow::compute::ExecNode* filter;
+      ARROW_ASSIGN_OR_RAISE(filter, arrow::compute::MakeExecNode("filter", 
+                            // plan
+                            plan.get(),
+                            // previous input
+                            {scan}, 
+                            //filter node options
+                            arrow::compute::FilterNodeOptions{filter_opt}));
+
+``ProjectNode``
+---------------
+
+:class:`ProjectNode` executes expressions on input batches and produces new batches. 
+Each expression will be evaluated against each batch which is pushed to this 
+node to produce a corresponding output column. This is exposed via 
+:class:`arrow::compute::ProjectNodeOptions` component which requires, 
+a :class:`arrow::compute::Expression`, names of the project columns (names are not provided, 
+the string representations of exprs will be used) and a boolean flag to determine 
+synchronous/asynchronous nature (by default asynchronous option is set to `true`). 
+
+Sample Expression for projection::
+
+    // a * 2 (multiply values in a column by 2)
+    arrow::compute::Expression a_times_2 = arrow::compute::call("multiply", 
+							{arrow::compute::field_ref("a"), arrow::compute::literal(2)});
+
+
+Creating a project node::
+
+    arrow::compute::ExecNode *project;
+        ARROW_ASSIGN_OR_RAISE(project, 
+            arrow::compute::MakeExecNode("project", 
+            // plan
+            plan.get(),
+            // previous node 
+            {scan},
+            // project node options 
+            arrow::compute::ProjectNodeOptions{{a_times_2}}));
+
+``ScalarAggregateNode``
```

Comment:
General comment: each of these sections can be made more consistent, in terms of wording and structure.

> (node name) does (X). It is constructed from (options class), which requires (parameters). (More description of parameters if necessary)
> 
> For example, to create a (node name) that does (specific example)::
>     (code example follows)